drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -111,12 +111,14 @@ struct nrf_usbd_ep_buf {
  * @param buf			Endpoint buffer
  * @param read_complete		A flag indicating that DMA read operation has been completed.
  * @param read_pending		A flag indicating that the Host has requested a data transfer.
+ * @param write_in_progress	A flag indicating that write operation has been scheduled.
  */
 struct nrf_usbd_ep_ctx {
 	struct nrf_usbd_ep_cfg cfg;
 	struct nrf_usbd_ep_buf buf;
 	volatile bool read_complete;
 	volatile bool read_pending;
+	volatile bool write_in_progress;
```

Comment:
Shouldn't you use `atomic_t` for those `volatile` values?

Reply:
volatile is misused but it may be that atomic is also not needed;
normally I would recommend to use atomic here to make sure operation read and write is done... atomically but zephyr implementation of atomics is expensive as it locks the irqs for all operations.